### PR TITLE
Add field `scheduled_points_honored` to the simulatio…

### PR DESCRIPTION
**Pull request by flomonster**
Used to detect train schedules that have one or more scheduled points that could not be fulfilled by the simulation.

### CODE REVIEW

**Comment by Castavo** [comment]
Why not add the full list of scheduled points which were not honored ? Such a list could be helpful for train imports debugging

**Comment by flomonster** [comment]
> Why not add the full list of scheduled points which were not honored ? Such a list could be helpful for train imports debugging

The front end will display the scheduled points that are not respected and points that do not respect the regularity margins. We imagine this will be done by the front end.
We also want core to forward warnings that should be stored and used for debugging.

File: core/src/main/kotlin/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractorV2.kt [hunk 1]
```diff
@@ -225,10 +227,26 @@ fun makeSimpleReportTrain(
             return@rdp abs(point.speed - projSpeed) * speedScaling +
                 abs(point.time - projTime) * timeScaling
         }
+
+    // Check if the scheduled points were respected
+    var scheduledPointsHonored = true
+    for (e in schedule) {
+        if (e.arrival == null) continue
+        var computed = envelopeStopWrapper.interpolateTotalTimeUS(e.pathOffset.distance.meters)
+        if (e.stopFor != null) computed -= e.stopFor.milliseconds * 1_000
```
Comment (eckter) [inline core/src/main/kotlin/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractorV2.kt]: We could do the computation with seconds there, I don't think we need high accuracy and juggling ms with µs is confusing

File: core/src/main/kotlin/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractorV2.kt [hunk 2]
```diff
@@ -225,10 +227,26 @@ fun makeSimpleReportTrain(
...
+        val expected = e.arrival.milliseconds * 1_000
+        // Check that the expected arrival time and the simulated one match ~1s
+        if ((computed - expected).absoluteValue > 1_000_000) {
+            scheduledPointsHonored = false
+            break
+        }
+    }
```
Comment (eckter) [inline core/src/main/kotlin/fr/sncf/osrd/standalone_sim/ScheduleMetadataExtractorV2.kt]: Nit: could be moved to a separate function for better readability (and control flow)

**Review by eckter — COMMENTED** [review]
Side note: technically all allowances can "fail" in a similar way, including standard allowances. Would we want to make the same kinds of checks in that context, or do we just ignore it?

It will eventually be included in the warning list anyway

**Review by woshilapin — APPROVED** [review]
Looks good for the `editoast` part.

**Comment by Castavo** [comment]
> The front end will display the scheduled points that are not respected and points that do not respect the regularity margins. We imagine this will be done by the front end.

If the list of points is computed by the front-end, then this is for train-import scripts only, why not leave the responsibility to the script to compare the results like the front would ?

**Review by Castavo — APPROVED** [review]
Code seems clear enough with its documentation 

It does what it's meant to do, tested it locally

**Review by leovalais — APPROVED** [review]
LGTM

**Review by eckter — APPROVED** [review]

**Pull request merged**